src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -2349,24 +2357,24 @@ FullWidthRepeat2:
 
                 ' // Check AM/PM
 
-                If CanGetCharAtOffset(Here) Then
-                    If PeekAheadChar(Here) = "A"c OrElse PeekAheadChar(Here) = FULLWIDTH_LATIN_CAPITAL_LETTER_A OrElse
-                        PeekAheadChar(Here) = "a"c OrElse PeekAheadChar(Here) = FULLWIDTH_LATIN_SMALL_LETTER_A Then
+                If TryPeekAheadChar(Here,ch) Then
```

Comment:
By "caching" the character from the peek, we don't pay the penalty of 3 addition function calls to `PeekAheadChar(Here)`